src/CodeViewer/ViewerTest.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <QApplication>
+#include <QFont>
+#include <QFontDatabase>
+#include <QFontMetrics>
+#include <array>
+#include <limits>
+
+#include "CodeViewer/Viewer.h"
+#include "gtest/gtest.h"
+
+namespace orbit_code_viewer {
+
+TEST(Viewer, DecimalDigitsFromNumber) {
+  EXPECT_EQ(DecimalDigitsFromNumber(0), 1);
+  EXPECT_EQ(DecimalDigitsFromNumber(1), 1);
+  EXPECT_EQ(DecimalDigitsFromNumber(9), 1);
+  EXPECT_EQ(DecimalDigitsFromNumber(10), 2);
+  EXPECT_EQ(DecimalDigitsFromNumber(123), 3);
+  EXPECT_EQ(DecimalDigitsFromNumber(123000), 6);
+
+  EXPECT_EQ(DecimalDigitsFromNumber(-1), 1);
```

Comment:
Ahhh. It seems we have a different understanding of a digit. Do you have a suggestion for a name that is less confusing?

Reply:
Sorry if I'm not being very clear. Please correct me if I'm wrong.

We are using this function to calculate the maximum space needed to draw a number (https://github.com/google/orbit/pull/1680/files#diff-5af26b9aee4a5a10b873c8289953c8e08704bcd67d825ce06632fafd50022023R51).

So, in my opinion, although we are probably not going to have any negative number there, the function should include the character "-". We can rename it as NumberCharactersFromNumber, and it should include "-". Am I being clearer now?

Reply:
My two cents:

I don't know a better name than digit and my (definitely not strong) preference would be to leave it as is. 

I also wouldn't mind karupayuns idea of changing it to characters. I mean to change the name and the behaviour. But then I am asking myself: why? Besides the test it's actually never used with negative numbers. And the test "defines" the exact behaviour, which is fine for me. 


What I am really wondering, is there a better without any custom function? Like maybe something in the standard library. I guess you could do `std::to_string(num).length()` but that is probably less performant.

Reply:
This function is bringing up way too much discussion. By factoring this code out into a function and adding a test, the intention way to generate less confusion, not more. I will rewrite that whole. Probably needs to be done anyway since this one test is broken. (Thanks Anton for discovering that.)

Reply:
I removed that function and also rewrote `DetermineLineNumberWidthInPixels`. The code is now less performant, but a lot shorter. PTAL!

Reply:
Thanks Pablo for raising this. It's probably better to go with the shorter version!